core/ecschema-metadata/src/Metadata/Format.ts

```diff
@@ -225,13 +300,34 @@ export class Format extends SchemaItem {
       itemElement.setAttribute("minWidth", this.minWidth.toString());
     if (undefined !== this.scientificType)
       itemElement.setAttribute("scientificType", this.scientificType);
+    if (undefined !== this.ratioType)
+      itemElement.setAttribute("ratioType", this.ratioType);
+    if (undefined !== this.ratioSeparator)
+      itemElement.setAttribute("ratioSeparator", this.ratioSeparator);
+    if (undefined !== this.ratioFormatType)
+      itemElement.setAttribute("ratioFormatType", this.ratioFormatType);
     if (undefined !== this.stationOffsetSize)
       itemElement.setAttribute("stationOffsetSize", this.stationOffsetSize.toString());
 
     const formatTraits = formatTraitsToArray(this.formatTraits);
     if (formatTraits.length > 0)
       itemElement.setAttribute("formatTraits", formatTraits.join("|"));
 
+    // Serialize ratioUnits
+    if (undefined !== this.ratioUnits && this.ratioUnits.length === 2) {
```

Comment:
This condition of `this.ratioUnits?.length === 2` is used in several places. Could be worth extracting to a function to give some semantic meaning and reduce errors?

Reply:
this is only used in 2 places, I know it can seem a lot larger because we also do this similar check against formatProps when deserializing but they are different interfaces so they can't all share a single function. And if we do define a single function, we run into no non-null assertion rules again (since the linter sucks) :(